### ISSUE TITLE
Static task observers always report Cancelled; return value is lost

### ISSUE THREAD

**Issue by drewcrawford** [drewcrawford/some_global_executor#2]
## Summary

`StaticExecutor::spawn_static` drops the `SpawnedStaticTask` wrapper before running the
future, so **every static task's observer reports `Cancelled` and the return value is
lost** — even though the future itself runs to completion. Task-locals (`TASK_ID`,
`TASK_LABEL`, `IS_CANCELLED`, ...) are never installed for static tasks either.

Both platform backends do it:

- `src/sys/stdlib.rs:312` — `spawned.into_future().await;`
- `src/sys/wasm/static_executor.rs:80` — `spawned.into_future().await; //swallow return value`

`into_future()` returns just the inner `F` and drops the rest of the wrapper, including
its `ObserverSender`. `ObserverSender::drop` sees the observation still `Pending` and
therefore marks it `Cancelled` (`some_executor/src/observer.rs:379`). The `Send` path
(`src/lib.rs:678`) does not have this problem — it passes the whole `spawned` through to
`spawn_internal`.

## Where

- Repo / branch: drewcrawford/some_global_executor @ main
- Commit: d033872
- Files: `src/sys/stdlib.rs:312`, `src/sys/wasm/static_executor.rs:80`

## Reproduce from a clean checkout

`Cargo.toml`:

```toml
[package]
name = "sge-repro"
version = "0.0.0"
edition = "2024"

[dependencies]
some_global_executor = { path = "../some_global_executor" }
some_executor = "0.7.1"
```

`src/main.rs`:

```rust
use some_executor::SomeExecutor;
use some_executor::observer::{FinishedObservation, Observer};
use some_executor::task::{Configuration, Task};

fn main() {
    let mut executor = some_global_executor::Executor::new("repro".to_string(), 1);

    let outer = Task::without_notifications(
        "outer".to_string(),
        Configuration::default(),
        async {
            // Hop onto the thread's static executor: static observers are !Send,
            // so they can only be awaited from a !Send task.
            let checker = Task::without_notifications(
                "checker".to_string(),
                Configuration::default(),
                async {
                    let inner = Task::without_notifications(
                        "inner-static".to_string(),
                        Configuration::default(),
                        async { 42i32 },
                    );
                    let observer = some_executor::thread_executor::thread_static_executor(|e| {
                        e.clone_box().spawn_static_objsafe(inner.into_objsafe_static())
                    });
                    match observer.await {
                        FinishedObservation::Ready(v) => {
                            let v = v.downcast::<i32>().expect("wrong type");
                            println!("RESULT: Ready({v})");
                        }
                        FinishedObservation::Cancelled => println!("RESULT: Cancelled"),
                    }
                },
            );
            checker.spawn_static_current();
        },
    );
    executor.spawn(outer).detach();
    executor.drain();
}
```

```bash
cargo run
```

## Expected vs actual

- Expected: `RESULT: Ready(42)`
- Actual: `RESULT: Cancelled`

The logs in the same run confirm the future did execute — `Static task 2 finished,
running_tasks=1` — so this is purely the observer channel being torn down early.

## Fix

`some_executor` 463bd46 ("Let external executors drive SpawnedStaticTask") added
`impl Future for SpawnedStaticTask` and made `SpawnedStaticTask::poll` public,
specifically because this workaround was the only reachable option before. Once a
`some_executor` release carrying that commit is out, the fix is:

```diff
-        self.spawn(
-            async {
-                spawned.into_future().await;
-            },
-            priority,
-            poll_after,
-        );
+        self.spawn(spawned, priority, poll_after);
```

in both backends (the wrapper is itself a `Future<Output = ()>`). Using
`SpawnedStaticTask::poll` directly instead would additionally let the executor populate
`TASK_STATIC_EXECUTOR` / `TASK_EXECUTOR` for the polled future, which the bare `Future`
impl leaves empty.

## Evidence already gathered

The existing static-executor tests all pass because they `detach()` the observer and
signal completion through an `AtomicBool` side channel
(e.g. `drain_waits_for_delayed_static_tasks`, `src/lib.rs:900`), so none of them ever
asserts on an observer's value. A test that does is what this needs.

Did NOT check: the wasm32 backend at runtime — it is the same code shape and the same
`into_future()` call, but I only reproduced on native.
